opentelemetry/src/trace/span_context.rs

```diff
@@ -94,89 +95,139 @@ impl fmt::LowerHex for TraceFlags {
     }
 }
 
-/// TraceId is an 16-byte value which uniquely identifies a given trace
-/// The actual `u128` value is wrapped in a tuple struct in order to leverage the newtype pattern
+/// A 16-byte value which identifies a given trace.
+///
+/// The id is valid if it contains at least one non-zero byte.
 #[cfg_attr(feature = "serialize", derive(Deserialize, Serialize))]
-#[derive(Clone, Debug, PartialEq, Eq, Copy, Hash)]
-pub struct TraceId(u128);
+#[derive(Clone, PartialEq, Eq, Copy, Hash)]
+pub struct TraceId(pub(crate) u128);
 
 impl TraceId {
-    /// Construct a new invalid (zero-valued) TraceId
-    pub fn invalid() -> Self {
-        TraceId(0)
+    /// Invalid trace id
+    pub const INVALID: TraceId = TraceId(0);
+
+    /// Create a trace id from its representation as a byte array.
+    pub const fn from_bytes(bytes: [u8; 16]) -> Self {
+        TraceId(u128::from_be_bytes(bytes))
     }
 
-    /// Convert from u128 to TraceId
-    pub fn from_u128(item: u128) -> Self {
```

Comment:
Hmm, does the spec mandate that we not have a conversion from `u128`? That seems a little silly. I think having an `impl From<u128> for TraceId` could make sense.

Reply:
They were trying to avoid specifying an endianness I believe. I don't have strong feelings about it either way.

Reply:
Trying to get the api slimmed down to just what the spec mandates so we can get a 1.0 release, can always bring this back if there is demand but currently this does bring the exporters more in line with other language's implementations in terms of relying more on opaque byte arrays.

Reply:
Fair enough.